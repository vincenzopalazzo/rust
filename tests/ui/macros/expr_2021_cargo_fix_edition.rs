//@  compile-flags: --edition=2021
// This test ensures that expr_2021 is not allowed on pre-2021 editions
#![deny(edition_2024_expr_fragment_specifier)]

macro_rules! m {
    ($e:expr) => { //~ ERROR: the `expr` fragment specifier will accept more expressions in the 2024 edition.
       //~^ WARN: this changes meaning in Rust 2024
        $e
    };
}

fn main() {
    m!(());
}

