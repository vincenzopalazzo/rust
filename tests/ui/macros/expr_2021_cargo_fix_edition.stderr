error: the `expr` fragment specifier will accept more expressions in the 2024 edition.
  --> $DIR/expr_2021_cargo_fix_edition.rs:6:9
   |
LL |     ($e:expr) => {
   |         ^^^^
   |
   = warning: this changes meaning in Rust 2024
   = note: for more information, see issue #123742 <https://github.com/rust-lang/rust/issues/123742>
note: the lint level is defined here
  --> $DIR/expr_2021_cargo_fix_edition.rs:3:9
   |
LL | #![deny(edition_2024_expr_fragment_specifier)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: to keep the existing behavior, use the `expr_2021` fragment specifier.
   |
LL |     ($e:expr_2021) => {
   |         ~~~~~~~~~

error: aborting due to 1 previous error

